src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -706,8 +706,8 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
             bool yielding = false;
             foreach (TSource element in source)
             {
-                if (!yielding && !predicate(element)) yielding = true;
                 if (yielding) yield return element;
+                else if (!predicate(element)) yielding = true;
```

Comment:
This is broken.  It misses yielding the first element that doesn't match the predicate.